lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +222,27 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        url = ''
+        for div in article_bs.find(class_="card-body").find_all("div", {"class": "title"}):
+            for link in div.select("a"):
+                url = link['href']
+        return self.url_pattern + url
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        urls = []
+        while len(urls) < self.config.get_num_articles():
+            for url in self.get_search_urls():
+                response = make_request(url, self.config)
+
+                if not response.ok:
+                    continue
+
+                soup = BeautifulSoup(response.text, 'lxml')
+                urls.append(self._extract_url(soup))
+        self.urls.extend(urls)
```

Comment:
why not changing `self.urls` directly? What is the reason of first using intermediate variable to store links?

Reply:
changed it